x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curation/results/utils.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Result } from '../../../result/types';
+import { CurationResult } from '../../types';
+
+/**
+ * The `promoted` and `hidden` keys from the internal curations endpoints
+ * currently return a document data structure that our Result component can't
+ * correctly parse - we need to attempt to naively transform the data in order
+ * to display it in a Result
+ *
+ * TODO: Ideally someday we can update our internal curations endpoint to return
+ * the same Result-ready data structure that the `organic` endpoint uses, and
+ * remove this file when that happens
+ */
+
+export const convertToResultFormat = (document: CurationResult): Result => {
+  const result = {} as Result;
+
+  // Convert `key: 'value'` into `key: { raw: 'value' }`
+  Object.entries(document).forEach(([key, value]) => {
+    result[key] = {
+      raw: value,
+      snippet: null, // Don't try to provide a snippet, we can't really guesstimate it
+    };
+  });
+
+  // Add the _meta obj needed by Result
+  result._meta = convertIdToMeta(document.id);
+
+  return result;
+};
+
+export const convertIdToMeta = (id: string): Result['_meta'] => {
+  const splitId = id.split('|');
+  const isMetaEngine = splitId.length > 1;
+
+  return isMetaEngine
+    ? {
+        engine: splitId[0],
+        id: splitId[1],
+      }
+    : ({ id } as Result['_meta']);
+  // Note: We're casting this as _meta even though `engine` is missing,
+  // since for source engines the engine shouldn't matter / be displayed,
+  // but if needed we could likely populate this from EngineLogic.values
```

Comment:
Also want to highlight this, non-meta engines don't need to show an `engine` in the Result card UI, so technically we don't need to try and get that data. If this ever changes we can work around it, but for now I opted to just cast the type 'incorrectly'.